### ISSUE TITLE
If anyone found track id duplicate over different objects in model.track() mode?

### ISSUE THREAD

**Issue by wayson20** [ultralytics/ultralytics#17866]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I use model.track() with default botsort.yaml, but i found the track id duplicate over different objects, if anyone else found ? And how can i solve this problem?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @wayson20, thank you for your interest in Ultralytics 🚀! We understand you're encountering an issue with `model.track()` and duplicate track IDs. Here's some information that can help address your concerns:

If this is a 🐛 Bug Report, please provide a full [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) (MRE), including the scripts, configurations, and any relevant data. This will allow us to better understand and address the issue. 

In the meantime, we recommend the following steps to ensure your setup is correct and up-to-date:

---

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not resolved in the latest version:

```bash
pip install -U ultralytics
```

## Troubleshooting Tips

1. If you're using `botsort.yaml`, ensure that any custom configurations or changes you’ve made align with the default parameters.
2. Review your input data to verify that the results are consistent with expected behavior.
3. Follow our [Docs on Tracking](https://docs.ultralytics.com/tasks/tracking/) to ensure you’re using `model.track()` optimally.

## Resources and Community

For more help, check out the Ultralytics community where you can get real-time assistance or discuss ideas:
- [Discord](https://discord.com/invite/ultralytics) 🎧
- [Discourse](https://community.ultralytics.com) 💬
- [Subreddit](https://reddit.com/r/Ultralytics) 🔗

---

## Environments

YOLO can be tested in the following verified environments for a smooth experience:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM - see [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI - see [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image** - see [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

---

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a> 

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

---

This is an automated response to help guide you on next steps! An Ultralytics engineer will also assist you soon. Let us know if you have additional information to share in the meantime 😊

**Comment by Y-T-G** [comment]
Are you using `persist=True`?

**Comment by wayson20** [comment]
Yes.发自我的 iPhone在 2024年11月29日，12:07，Mohammed Yasin ***@***.***> 写道：﻿
Are you using persist=True?

**Comment by Y-T-G** [comment]
Can you post an image of duplicate ids?

**Comment by wayson20** [comment]
2024-11-28 16:34:48,633 - INFO: target value: ('******_1_298462',
2024-11-28 16:37:44,515 - INFO: target value: ('******_1_298462'
I just get the info from logs, "298462" is the track id, the object is a truck run on the highway. It's strange over 3 minute the truck still on the camera. So, I guess another similar truck is assigned the same track id.

**Comment by Y-T-G** [comment]
You can try increasing this

https://github.com/ultralytics/ultralytics/blob/e60992214c91d3ba169965053af69d6eb233b45e/ultralytics/cfg/trackers/botsort.yaml#L9

**Comment by wayson20** [comment]
OK, thanks!

**Comment by glenn-jocher** [comment]
You're welcome! Let us know if you have any further questions.

**Comment by wayson20** [comment]
I set track_high_thresh=0.8, track_low_thresh=0.4, new_track_thresh=0.7, match_thresh=0.95, is that reasonable? How to evaluate the impact of these parameters?

**Comment by Y-T-G** [comment]
You should only need to increase match_thresh

**Comment by wayson20** [comment]
OK, I restore track_high_thresh, track_low_thresh, and new_track_thresh to their original values.

**Comment by glenn-jocher** [comment]
Restoring the values is a good step. To evaluate the impact of parameters like `match_thresh`, focus on tracking consistency and accuracy across frames. Gradually adjust and test changes while observing how effectively objects are tracked without ID switches or mismatches.

**Comment by wayson20** [comment]
OK, thanks!

**Comment by glenn-jocher** [comment]
You're welcome! Let us know if you encounter any more issues or need further assistance with your tracker configuration.
